### PR TITLE
Pass aria-label prop through as button attribute.

### PR DESCRIPTION
**Pull request by kid-icarus**
#### What does this PR do?

Previously we only used the a11yTitle prop for button aria labels. This is a little unintuitive to someone who is already familiar with a11y. Rather than deprecate the a11yTitle prop we allow for it in addition to aria-label.

#### Where should the reviewer start?

At the Button component.

#### What testing has been done on this PR?

Added a unit test to ensure the Button has no aXe a11y violations, manually tested to see the attribute rendered.

#### How should this be manually tested?

Create a Button, passing in aria-label. Then, check to see that the rendered button does in fact contain the given `aria-label` attribute.

#### Any background context you want to provide?

I created a `Button` and gave it an `aria-label`. TypeScript allowed this, which tripped me up. An alternative here is to just ensure that the [ButtonExtendedProps](https://github.com/kid-icarus/grommet/blob/cea630132b637c223dd155292afdcd5399afbdce/src/js/components/Button/index.d.ts#L45) disallows `aria-label`, which I tried, but was unable to get it to work properly by adding `aria-label` to the list of Omitted keys.

#### What are the relevant issues?

Someone mentioned this [here](https://github.com/grommet/grommet/issues/2529)

#### Should this PR be mentioned in the release notes?

It is technically modifying the API, a minor bump may be warranted.

#### Is this change backwards compatible or is it a breaking change?

It is backwards compatible

### CODE REVIEW

**Comment by ShimiSun** [comment]
quick question, have you tried to pass `aria-label` via `rest` and it didn't work?

**Comment by kid-icarus** [comment]
Oh good point @ShimiSun, I did actually try that but we spread `...rest` _before_  setting `aria-label` to `a11yTitle,` so it is always overridden to undefined.

Perhaps the solution here just to put the spread at the end, happy to change this PR and test that solution.

**Comment by kid-icarus** [comment]
I actually just re-ordered the ...rest spread, and added a unit test to assert that the attribute is set correctly.

FYI, a strange issue here that I noticed: Even rendering a button with no aria-label, no a11yTitle, and no label still passes the aXe violation test, which seems incorrect to me.

**Comment by kid-icarus** [comment]
Additionally, I'd be happy to check the rest of the components here to ensure that aria-labels are appropriate spread onto the correct element for all components, we seem to have inconsistent ...rest spread positioning between components.

See [Anchor](https://github.com/kid-icarus/grommet/blob/3b93319c04e60f8caa6b9f3881131c855f09dc9f/src/js/components/Anchor/Anchor.js#L59) vs [Box](https://github.com/kid-icarus/grommet/blob/1c1b69df83cca1834a287a6287982c6ef1114809/src/js/components/Box/Box.js#L151)

File: src/js/components/Button/Button.js [hunk 1]
```diff
@@ -297,7 +297,6 @@ const Button = forwardRef(
     } else {
       styledButtonResult = (
         <StyledButton
-          {...rest}
           as={domTag}
           ref={ref}
           aria-label={a11yTitle}
```
Comment (ericsoderberghp) [inline src/js/components/Button/Button.js:302]:
```suggestion
          aria-label={aria-label || a11yTitle}
```

**Review by ericsoderberghp — COMMENTED** [review]
I think the approach suggested in this review is less destabilizing. If a caller passes in any property via rest that conflicts with something grommet is setting internally, it will break how Button works, such as `hasIcon`.

**Comment by kid-icarus** [comment]
This makes sense to me @ericsoderberghp and I think it may be beneficial to standardize (perhaps a lint rule exists for this purpose) spreading first so as to not override internally set properties.

One could argue that a TypeSript would prevent people from passing props that aren't allowed, but alas this is JavaScript world and propTypes only go so far.

**Review by ericsoderberghp — APPROVED** [review]

File: src/js/components/Button/__tests__/Button-test.js [hunk 2]
```diff
@@ -26,6 +26,23 @@ describe('Button', () => {
...
+    );
+
+    const button = container.querySelector('button');
+    expect(button).toBeTruthy();
+    expect(button.getAttribute('aria-label')).toEqual('Test Label');
+
+    fireEvent.click(getByText('Test'));
+    const results = await axe(container);
+    expect(results).toHaveNoViolations();
+  });
```
Comment (ShimiSun) [inline src/js/components/Button/__tests__/Button-test.js:44]: Where are the snapshots created for this test that are indicating that `aria-label` is applied as expected?

Comment (kid-icarus) [reply src/js/components/Button/__tests__/Button-test.js:44]: There are no snapshots, I'm just manually asserting that the attribute is correctly set. If you find a snapshot valuable, I'm happy to add it.

Comment (kid-icarus) [reply src/js/components/Button/__tests__/Button-test.js]: @ShimiSun I've added a snapshot that looks at both `a11yTitle` and `aria-label` props.

**Review by ShimiSun — APPROVED** [review]

**Comment by ShimiSun** [comment]
Thank you for your contribution!

**Pull request merged**